4046 - Add AssumeRole and update appropriate tests (#4132)
kubernetes-sigs/cluster-api-provider-aws: docs/book/src/topics/using-clusterawsadm-to-fulfill-prerequisites.md

  ...
```

#### Cross Account Role Assumption

> CAPA, by default, does not provide the necessary permissions to allow cross account role assumption which can be used to manage clusters in other environments.  This is documented [here](multitenancy.md#necessary-permissions-for-assuming-a-role).  The 'sts:AssumeRole' permissions can be added via the following configuration on the manager account configuration:

Reviewer: ```suggestion
CAPA, by default, does not provide the necessary permissions to allow cross-account role assumption, which can be used to manage clusters in other environments.  This is documented [here](multitenancy.md#necessary-permissions-for-assuming-a-role).  The 'sts:AssumeRole' permissions can be added via the following configuration on the manager account configuration:
```

Reviewer: Also, this linking will not work in the rendered documentation.

```yaml
apiVersion: bootstrap.aws.infrastructure.cluster.x-k8s.io/v1beta1
kind: AWSIAMConfiguration
spec:
  ...

>   AllowAssumeRole: true

Reviewer: This should be `allowAssumeRole`.

  ...
```

> The above will allow the controller to have the necessary permissions needed in order for it to manage clusters in other accounts through the use of the AWSClusterRoleIdentity. Please note, the above should only be applied to the account where CAPA is running.  To allow CAPA to assume the roles in the managed/target accounts, the following configuration needs to be used:

Reviewer: ```suggestion
The above will give the controller the necessary permissions to manage clusters in other accounts using the AWSClusterRoleIdentity. Please note the above should only be applied to the account where CAPA is running.  To allow CAPA to assume the roles in the managed/target accounts, the following configuration needs to be used:
```
